From e14f06fc9aa81c8edd529b6af0ad092d5684abf0 Mon Sep 17 00:00:00 2001
From: Bryan Brattlof <bb@ti.com>
Date: Tue, 1 Feb 2022 14:37:31 -0600
Subject: [PATCH 5/8] soc: j721s2: add processor description information

This provides information of Processor and Host IDs that are permitted
in the j721s2 SoC. These Processor IDs represent an actual physical
processor on the SoC.

Signed-off-by: Bryan Brattlof <bb@ti.com>
---
 Makefile                            |  3 +-
 common/socinfo.c                    |  3 ++
 include/tisci.h                     |  2 +-
 soc/j721s2/j721s2_processors_info.c | 49 +++++++++++++++++++++++++++++
 soc/j721s2/j721s2_processors_info.h | 41 ++++++++++++++++++++++++
 5 files changed, 96 insertions(+), 2 deletions(-)
 create mode 100644 soc/j721s2/j721s2_processors_info.c
 create mode 100644 soc/j721s2/j721s2_processors_info.h

diff --git a/Makefile b/Makefile
index ed3f247..e1e8137 100644
--- a/Makefile
+++ b/Makefile
@@ -121,7 +121,8 @@ AM62XSOURCES =\
 J721S2SOURCES =\
 	       soc/j721s2/j721s2_devices_info.c \
 	       soc/j721s2/j721s2_host_info.c \
-	       soc/j721s2/j721s2_clocks_info.c
+	       soc/j721s2/j721s2_clocks_info.c \
+	       soc/j721s2/j721s2_processors_info.c
 
 COMMONOBJECTS=	$(COMMONSOURCES:.c=.o)
 AM65XOBJECTS=	$(AM65XSOURCES:.c=.o)
diff --git a/common/socinfo.c b/common/socinfo.c
index 04f54bf..d2d2bd1 100644
--- a/common/socinfo.c
+++ b/common/socinfo.c
@@ -76,6 +76,7 @@
 #include <soc/j721s2/j721s2_devices_info.h>
 #include <soc/j721s2/j721s2_host_info.h>
 #include <soc/j721s2/j721s2_clocks_info.h>
+#include <soc/j721s2/j721s2_processors_info.h>
 
 /* Assuming these addresses and definitions stay common across K3 devices */
 #define CTRLMMR_WKUP_JTAG_DEVICE_ID	0x43000018
@@ -157,6 +158,8 @@ static void j721s2_init(void)
 {
 	struct ti_sci_info *sci_info = &soc_info.sci_info;
 
+	sci_info->processors_info = j721s2_processors_info;
+	sci_info->num_processors = J721S2_MAX_PROCESSORS_IDS;
 	sci_info->clocks_info = j721s2_clocks_info;
 	sci_info->num_clocks = J721S2_MAX_CLOCKS;
 	sci_info->host_info = j721s2_host_info;
diff --git a/include/tisci.h b/include/tisci.h
index eabc2a7..f185771 100644
--- a/include/tisci.h
+++ b/include/tisci.h
@@ -67,7 +67,7 @@ struct ti_sci_processors_info {
 	uint32_t dev_id;
 	uint32_t clk_id;
 	uint32_t processor_id;
-	char name[20];
+	char name[25];
 };
 
 struct ti_sci_devices_info {
diff --git a/soc/j721s2/j721s2_processors_info.c b/soc/j721s2/j721s2_processors_info.c
new file mode 100644
index 0000000..6a1ce7d
--- /dev/null
+++ b/soc/j721s2/j721s2_processors_info.c
@@ -0,0 +1,49 @@
+/*
+ * J721S2 Processor Info
+ *
+ * Copyright (C) 2022 Texas Instruments Incorporated - https://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <tisci.h>
+#include <socinfo.h>
+
+struct ti_sci_processors_info j721s2_processors_info[] = {
+	[0] = {202, 0, 0x20, "A72SS0_CORE0"},
+	[1] = {203, 0, 0x21, "A72SS0_CORE1"},
+	[2] = {8,   0, 0x30, "COMPUTE_CLUSTER0_C71SS0_0"},
+	[3] = {11,  0, 0x31, "COMPUTE_CLUSTER0_C71SS1_0"},
+	[4] = {284, 0, 0x01, "MCU_R5FSS0_CORE0"},
+	[5] = {285, 0, 0x02, "MCU_R5FSS0_CORE1"},
+	[6] = {279, 0, 0x06, "R5FSS0_CORE0"},
+	[7] = {280, 0, 0x07, "R5FSS0_CORE1"},
+	[8] = {281, 0, 0x08, "R5FSS1_CORE0"},
+	[9] = {282, 0, 0x09, "R5FSS1_CORE1"},
+};
diff --git a/soc/j721s2/j721s2_processors_info.h b/soc/j721s2/j721s2_processors_info.h
new file mode 100644
index 0000000..c430a4d
--- /dev/null
+++ b/soc/j721s2/j721s2_processors_info.h
@@ -0,0 +1,41 @@
+/*
+ * J721S2 Processor Info
+ *
+ * Copyright (C) 2022 Texas Instruments Incorporated - https://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __J721S2_PROCESSOR_INFO_H
+#define __J721S2_PROCESSOR_INFO_H
+
+#define J721S2_MAX_PROCESSORS_IDS		10
+extern struct ti_sci_processors_info j721s2_processors_info[];
+
+#endif /* __J721S2_PROCESSOR_INFO_H */
\ No newline at end of file
-- 
2.17.1

